src/core/src/FabricMaterial.cpp

```diff
@@ -0,0 +1,351 @@
+        attributes.addAttribute(FabricTypes::_relationships_outputName, FabricTokens::_relationships_outputName);
+        attributes.addAttribute(FabricTypes::primvars, FabricTokens::primvars);
+        attributes.addAttribute(FabricTypes::MaterialNetwork, FabricTokens::MaterialNetwork);
+        attributes.addAttribute(FabricTypes::_cesium_tilesetId, FabricTokens::_cesium_tilesetId);
+        attributes.addAttribute(FabricTypes::_cesium_tileId, FabricTokens::_cesium_tileId);
+
+        attributes.createAttributes(surfacePathFabric);
+
+        sip.setArrayAttributeSize(surfacePathFabric, FabricTokens::_nodePaths, 1);
+        sip.setArrayAttributeSize(surfacePathFabric, FabricTokens::_relationships_inputId, 0);
+        sip.setArrayAttributeSize(surfacePathFabric, FabricTokens::_relationships_outputId, 0);
+        sip.setArrayAttributeSize(surfacePathFabric, FabricTokens::_relationships_inputName, 0);
+        sip.setArrayAttributeSize(surfacePathFabric, FabricTokens::_relationships_outputName, 0);
+        sip.setArrayAttributeSize(surfacePathFabric, FabricTokens::primvars, 0);
+
+        auto nodePathsFabric = sip.getArrayAttributeWr<uint64_t>(surfacePathFabric, FabricTokens::_nodePaths);
+
+        nodePathsFabric[0] = shaderPathFabricUint64;
+    }
+
+    // Shader
+    {
+        const auto shaderPathFabric = carb::flatcache::Path(carb::flatcache::asInt(shaderPath));
+        sip.createPrim(shaderPathFabric);
+
+        FabricAttributesBuilder attributes;
+
+        // clang-format off
+        attributes.addAttribute(FabricTypes::info_id, FabricTokens::info_id);
+        attributes.addAttribute(FabricTypes::info_sourceAsset_subIdentifier, FabricTokens::info_sourceAsset_subIdentifier);
+        attributes.addAttribute(FabricTypes::_paramColorSpace, FabricTokens::_paramColorSpace);
+        attributes.addAttribute(FabricTypes::_parameters, FabricTokens::_parameters);
+        attributes.addAttribute(FabricTypes::diffuse_color_constant, FabricTokens::diffuse_color_constant);
+        attributes.addAttribute(FabricTypes::metallic_constant, FabricTokens::metallic_constant);
+        attributes.addAttribute(FabricTypes::reflection_roughness_constant, FabricTokens::reflection_roughness_constant);
+        attributes.addAttribute(FabricTypes::specular_level, FabricTokens::specular_level);
+        attributes.addAttribute(FabricTypes::Shader, FabricTokens::Shader);
+        attributes.addAttribute(FabricTypes::_cesium_tilesetId, FabricTokens::_cesium_tilesetId);
+        attributes.addAttribute(FabricTypes::_cesium_tileId, FabricTokens::_cesium_tileId);
+        // clang-format on
+
+        if (hasBaseColorTexture) {
+            attributes.addAttribute(FabricTypes::diffuse_texture, FabricTokens::diffuse_texture);
+        }
+
+        attributes.createAttributes(shaderPathFabric);
+
+        const size_t paramColorSpaceSize = hasBaseColorTexture ? 2 : 0;
```

Comment:
You may want to add some comments here to explain what all the magic numbers are. It may be obvious now, but it may not be obvious in 6+ months.